### PR TITLE
benchbot: use specified binary deploying for component

### LINKED ISSUE #43: benchbot: use specified binary deploying for component
## Description

Benchbot uses binaries(usually download from inner fileserver) to deploy a test cluster, so Naglfar should enhance the TCT part, means adding the below struct:

``` go
type TiDBClusterBinaryURL struct {
	TiDB string `json:"tidb"`
	TiKV string `json:"tikv"`
	PD string `json:"pd"`
}
```

### PR DESCRIPTION
**Pull request by mahjonp**
Signed-off-by: mahjonp <junpeng.man@gmail.com>

close #43

### CODE REVIEW

File: pkg/tiup/deploy.go [hunk 1]
```diff
@@ -301,7 +309,85 @@ func (c *ClusterManager) startCluster(clusterName string) error {
...
+		commands = append(commands, c.GenUnzipCommand(path.Base(u.Path), "components"))
+		commands = append(commands, fmt.Sprintf("rm -rf components/%s", component.componentName))
+		commands = append(commands, fmt.Sprintf("mv components/bin/%s components/%s", component.componentName, component.componentName))
+	}
+	commands = append(commands, "cd components && tar zcf patch.tar.gz "+strings.Join(patchComponentNames, " "))
+	for _, component := range patchComponentNames {
+		commands = append(commands, fmt.Sprintf("/root/.tiup/bin/tiup cluster patch %s patch.tar.gz -R %s",
+			clusterName, strings.Split(component, "-server")[0]))
+	}
+	cmd := fmt.Sprintf(`flock -n /tmp/naglfar.tiup.lock -c "%s"`, strings.Join(commands, "\n"))
```
Comment (mahjonp) [inline pkg/tiup/deploy.go:372]: Use flock here to disallow performing multi patches concurrently

File: pkg/tiup/deploy.go [hunk 2]
```diff
@@ -301,7 +309,85 @@ func (c *ClusterManager) startCluster(clusterName string) error {
...
+			"command", cmd,
+			"stdout", stdStr,
+			"stderr", errStr)
+		return fmt.Errorf("patch cluster failed(%s): %s", err, errStr)
+	}
+	return nil
+}
+
+func (c *ClusterManager) GenUnzipCommand(filePath string, destDir string) string {
+	if strings.HasPrefix(filePath, ".zip") {
```
Comment (Hexilee) [inline pkg/tiup/deploy.go:386]: HasSuffix?

**Review by Hexilee — CHANGES_REQUESTED** [review]
Rest LGTM

File: pkg/tiup/deploy.go (hunk 2 above)
Comment (mahjonp) [reply pkg/tiup/deploy.go]: solved on 93f8ece

**Review by Hexilee — APPROVED** [review]
LGTM

**Pull request merged**